### ISSUE TITLE
Question about the Training Procedure for the SVTRv2 Chinese Recognition Model

### ISSUE THREAD

**Issue by yuyuyu864** [Topdu/OpenOCR#209]
Hello, I would like to ask about the complete training procedure for the SVTRv2 Chinese text recognition model.
In the readme, the English model is trained in two stages: first, svtrv2_rctc.yml is used to train the first-stage model and obtain ./output/rec/u14m_filter/svtrv2_rctc/best.pth; then, svtrv2_smtr_gtc_rctc.yml loads this checkpoint for the second-stage training.Does the Chinese model also require two-stage training? If so, which configuration file should be used for the first stage? Should I first train a Chinese RCTC/CTC model on the BCTR training set, or can I directly load the English checkpoint ./output/rec/u14m_filter/svtrv2_rctc/best.pth?
Thank you very much!

**Comment by Topdu** [comment]
The Chinese model does not undergo the RCTC training phase; it only undergoes the final training phase and is trained from scratch without loading the English model.

**Comment by yuyuyu864** [comment]
> 中国型号不经过RCTC训练阶段;它只经历最终的训练阶段，并且从零开始训练，无需加载英文模型。

Thank you very much for your previous reply and helpful guidance. I really appreciate your time and support.
I am currently reproducing the SVTRv2 experiments on the Chinese BCTR benchmark, and I encountered a question regarding the training data statistics. I would like to ask for your advice.In my experiment, I used the four Chinese training subsets,After initializing the training dataloader, the bucket statistics printed in my log are:
[2026/07/11 04:50:14] openrec INFO: ----------- Config -----------
[2026/07/11 04:50:14] openrec INFO: Architecture : 
[2026/07/11 04:50:14] openrec INFO:     Decoder : 
[2026/07/11 04:50:14] openrec INFO:         ctc_decoder : 
[2026/07/11 04:50:14] openrec INFO:             name : RCTCDecoder
[2026/07/11 04:50:14] openrec INFO:         detach : False
[2026/07/11 04:50:14] openrec INFO:         gtc_decoder : 
[2026/07/11 04:50:14] openrec INFO:             ds : True
[2026/07/11 04:50:14] openrec INFO:             max_len : 25
[2026/07/11 04:50:14] openrec INFO:             name : SMTRDecoder
[2026/07/11 04:50:14] openrec INFO:             next_mode : True
[2026/07/11 04:50:14] openrec INFO:             num_layer : 1
[2026/07/11 04:50:14] openrec INFO:             sub_str_len : 5
[2026/07/11 04:50:14] openrec INFO:         infer_gtc : True
[2026/07/11 04:50:14] openrec INFO:         name : GTCDecoder
[2026/07/11 04:50:14] openrec INFO:     Encoder : 
[2026/07/11 04:50:14] openrec INFO:         depths : [6, 6, 6]
[2026/07/11 04:50:14] openrec INFO:         dims : [128, 256, 384]
[2026/07/11 04:50:14] openrec INFO:         feat2d : True
[2026/07/11 04:50:14] openrec INFO:         last_stage : False
[2026/07/11 04:50:14] openrec INFO:         local_k : [[5, 5], [5, 5], [-1, -1]]
[2026/07/11 04:50:14] openrec INFO:         mixer : [['Conv', 'Conv', 'Conv', 'Conv', 'Conv', 'Conv'], ['Conv', 'Conv', 'FGlobal', 'Global', 'Global', 'Global'], ['Global', 'Global', 'Global', 'Global', 'Global', 'Global']]
[2026/07/11 04:50:14] openrec INFO:         name : SVTRv2LNConvTwo33
[2026/07/11 04:50:14] openrec INFO:         num_heads : [4, 8, 12]
[2026/07/11 04:50:14] openrec INFO:         sub_k : [[1, 1], [2, 1], [-1, -1]]
[2026/07/11 04:50:14] openrec INFO:         use_pos_embed : False
[2026/07/11 04:50:14] openrec INFO:     Transform : None
[2026/07/11 04:50:14] openrec INFO:     algorithm : SVTRv2
[2026/07/11 04:50:14] openrec INFO:     in_channels : 3
[2026/07/11 04:50:14] openrec INFO:     model_type : rec
[2026/07/11 04:50:14] openrec INFO: Eval : 
[2026/07/11 04:50:14] openrec INFO:     dataset : 
[2026/07/11 04:50:14] openrec INFO:         data_dir_list : ['/opt/data/private/data/benchmark_bctr/benchmark_bctr_test/document_test', '/opt/data/private/data/handwriting1/test', '/opt/data/private/data/benchmark_bctr/benchmark_bctr_test/scene_test', '/opt/data/private/data/benchmark_bctr/benchmark_bctr_test/web_test']
[2026/07/11 04:50:14] openrec INFO:         ds_width : True
[2026/07/11 04:50:14] openrec INFO:         name : RatioDataSetTVResize
[2026/07/11 04:50:14] openrec INFO:         padding : False
[2026/07/11 04:50:14] openrec INFO:         transforms : 
[2026/07/11 04:50:14] openrec INFO:             DecodeImagePIL : 
[2026/07/11 04:50:14] openrec INFO:                 img_mode : RGB
[2026/07/11 04:50:14] openrec INFO:             GTCLabelEncode : 
[2026/07/11 04:50:14] openrec INFO:                 character_dict_path : ./tools/utils/ppocr_keys_v1.txt
[2026/07/11 04:50:14] openrec INFO:                 gtc_label_encode : 
[2026/07/11 04:50:14] openrec INFO:                     name : ARLabelEncode
[2026/07/11 04:50:14] openrec INFO:                 max_text_length : 25
[2026/07/11 04:50:14] openrec INFO:                 use_space_char : False
[2026/07/11 04:50:14] openrec INFO:             KeepKeys : 
[2026/07/11 04:50:14] openrec INFO:                 keep_keys : ['image', 'label', 'length', 'ctc_label', 'ctc_length']
[2026/07/11 04:50:14] openrec INFO:     loader : 
[2026/07/11 04:50:14] openrec INFO:         batch_size_per_card : 256
[2026/07/11 04:50:14] openrec INFO:         drop_last : False
[2026/07/11 04:50:14] openrec INFO:         max_ratio : 8
[2026/07/11 04:50:14] openrec INFO:         num_workers : 4
[2026/07/11 04:50:14] openrec INFO:         shuffle : False
[2026/07/11 04:50:14] openrec INFO:     sampler : 
[2026/07/11 04:50:14] openrec INFO:         divided_factor : [4, 16]
[2026/07/11 04:50:14] openrec INFO:         first_bs : 256
[2026/07/11 04:50:14] openrec INFO:         fix_bs : False
[2026/07/11 04:50:14] openrec INFO:         is_training : False
[2026/07/11 04:50:14] openrec INFO:         name : RatioSampler
[2026/07/11 04:50:14] openrec INFO:         scales : [[128, 32]]
[2026/07/11 04:50:14] openrec INFO: Global : 
[2026/07/11 04:50:14] openrec INFO:     accumulation_steps : 4
[2026/07/11 04:50:14] openrec INFO:     cal_metric_during_train : False
[2026/07/11 04:50:14] openrec INFO:     character_dict_path : ./tools/utils/ppocr_keys_v1.txt
[2026/07/11 04:50:14] openrec INFO:     checkpoints : None
[2026/07/11 04:50:14] openrec INFO:     device : gpu
[2026/07/11 04:50:14] openrec INFO:     distributed : False
[2026/07/11 04:50:14] openrec INFO:     epoch_num : 100
[2026/07/11 04:50:14] openrec INFO:     eval_batch_step : [0, 2000]
[2026/07/11 04:50:14] openrec INFO:     eval_epoch_step : [0, 1]
[2026/07/11 04:50:14] openrec INFO:     infer_img : None
[2026/07/11 04:50:14] openrec INFO:     log_smooth_window : 20
[2026/07/11 04:50:14] openrec INFO:     max_text_length : 25
[2026/07/11 04:50:14] openrec INFO:     output_dir : ./output/rec/ch/svtrv2_smtr_gtc_rctc_ch1
[2026/07/11 04:50:14] openrec INFO:     pretrained_model : None
[2026/07/11 04:50:14] openrec INFO:     print_batch_step : 10
[2026/07/11 04:50:14] openrec INFO:     save_epoch_step : [15, 10]
[2026/07/11 04:50:14] openrec INFO:     save_res_path : ./output/rec/ch/predicts_svtrv2_smtr_gtc_rctc_ch1.txt
[2026/07/11 04:50:14] openrec INFO:     use_amp : True
[2026/07/11 04:50:14] openrec INFO:     use_space_char : False
[2026/07/11 04:50:14] openrec INFO:     use_tensorboard : False
[2026/07/11 04:50:14] openrec INFO: LRScheduler : 
[2026/07/11 04:50:14] openrec INFO:     cycle_momentum : False
[2026/07/11 04:50:14] openrec INFO:     name : OneCycleLR
[2026/07/11 04:50:14] openrec INFO:     warmup_epoch : 5
[2026/07/11 04:50:14] openrec INFO: Loss : 
[2026/07/11 04:50:14] openrec INFO:     ctc_weight : 0.1
[2026/07/11 04:50:14] openrec INFO:     gtc_loss : 
[2026/07/11 04:50:14] openrec INFO:         name : SMTRLoss
[2026/07/11 04:50:14] openrec INFO:     name : GTCLoss
[2026/07/11 04:50:14] openrec INFO: Metric : 
[2026/07/11 04:50:14] openrec INFO:     main_indicator : acc
[2026/07/11 04:50:14] openrec INFO:     name : RecGTCMetric
[2026/07/11 04:50:14] openrec INFO: Optimizer : 
[2026/07/11 04:50:14] openrec INFO:     filter_bias_and_bn : True
[2026/07/11 04:50:14] openrec INFO:     lr : 0.00065
[2026/07/11 04:50:14] openrec INFO:     name : AdamW
[2026/07/11 04:50:14] openrec INFO:     weight_decay : 0.05
[2026/07/11 04:50:14] openrec INFO: PostProcess : 
[2026/07/11 04:50:14] openrec INFO:     character_dict_path : ./tools/utils/ppocr_keys_v1.txt
[2026/07/11 04:50:14] openrec INFO:     gtc_label_decode : 
[2026/07/11 04:50:14] openrec INFO:         name : SMTRLabelDecode
[2026/07/11 04:50:14] openrec INFO:         next_mode : True
[2026/07/11 04:50:14] openrec INFO:     name : GTCLabelDecode
[2026/07/11 04:50:14] openrec INFO:     use_space_char : False
[2026/07/11 04:50:14] openrec INFO: Train : 
[2026/07/11 04:50:14] openrec INFO:     dataset : 
[2026/07/11 04:50:14] openrec INFO:         data_dir_list : ['/opt/data/private/data/benchmark_bctr/benchmark_bctr_train/document_train', '/opt/data/private/data/handwriting1/train', '/opt/data/private/data/benchmark_bctr/benchmark_bctr_train/scene_train', '/opt/data/private/data/benchmark_bctr/benchmark_bctr_train/web_train']
[2026/07/11 04:50:14] openrec INFO:         ds_width : True
[2026/07/11 04:50:14] openrec INFO:         name : RatioDataSetTVResize
[2026/07/11 04:50:14] openrec INFO:         padding : False
[2026/07/11 04:50:14] openrec INFO:         transforms : 
[2026/07/11 04:50:14] openrec INFO:             DecodeImagePIL : 
[2026/07/11 04:50:14] openrec INFO:                 img_mode : RGB
[2026/07/11 04:50:14] openrec INFO:             PARSeqAugPIL : None
[2026/07/11 04:50:14] openrec INFO:             GTCLabelEncode : 
[2026/07/11 04:50:14] openrec INFO:                 character_dict_path : ./tools/utils/ppocr_keys_v1.txt
[2026/07/11 04:50:14] openrec INFO:                 gtc_label_encode : 
[2026/07/11 04:50:14] openrec INFO:                     name : SMTRLabelEncode
[2026/07/11 04:50:14] openrec INFO:                     sub_str_len : 5
[2026/07/11 04:50:14] openrec INFO:                 max_text_length : 25
[2026/07/11 04:50:14] openrec INFO:                 use_space_char : False
[2026/07/11 04:50:14] openrec INFO:             KeepKeys : 
[2026/07/11 04:50:14] openrec INFO:                 keep_keys : ['image', 'label', 'label_subs', 'label_next', 'length_subs', 'label_subs_pre', 'label_next_pre', 'length_subs_pre', 'length', 'ctc_label', 'ctc_length']
[2026/07/11 04:50:14] openrec INFO:     loader : 
[2026/07/11 04:50:14] openrec INFO:         batch_size_per_card : 256
[2026/07/11 04:50:14] openrec INFO:         drop_last : True
[2026/07/11 04:50:14] openrec INFO:         max_ratio : 8
[2026/07/11 04:50:14] openrec INFO:         num_workers : 4
[2026/07/11 04:50:14] openrec INFO:         shuffle : True
[2026/07/11 04:50:14] openrec INFO:     sampler : 
[2026/07/11 04:50:14] openrec INFO:         divided_factor : [4, 16]
[2026/07/11 04:50:14] openrec INFO:         first_bs : 256
[2026/07/11 04:50:14] openrec INFO:         fix_bs : False
[2026/07/11 04:50:14] openrec INFO:         is_training : True
[2026/07/11 04:50:14] openrec INFO:         name : RatioSampler
[2026/07/11 04:50:14] openrec INFO:         scales : [[128, 32]]
[2026/07/11 04:50:14] openrec INFO: config : configs/rec/svtrv2/svtrv2_smtr_gtc_rctc_ch.yml
[2026/07/11 04:50:14] openrec INFO: eval : True
[2026/07/11 04:50:14] openrec INFO: filename : svtrv2_smtr_gtc_rctc_ch
[2026/07/11 04:50:14] openrec INFO: local_rank : 0
[2026/07/11 04:50:14] openrec INFO: ---------------------------------------------
[2026/07/11 04:50:14] openrec INFO: Initialize indexs of datasets:/opt/data/private/data/benchmark_bctr/benchmark_bctr_train/document_train
[2026/07/11 04:50:14] openrec INFO: Initialize indexs of datasets:/opt/data/private/data/handwriting1/train
[2026/07/11 04:50:14] openrec INFO: Initialize indexs of datasets:/opt/data/private/data/benchmark_bctr/benchmark_bctr_train/scene_train
[2026/07/11 04:50:14] openrec INFO: Initialize indexs of datasets:/opt/data/private/data/benchmark_bctr/benchmark_bctr_train/web_train
[2026/07/11 04:50:47] openrec INFO: 0
[2026/07/11 04:50:47] openrec INFO: 209907
[2026/07/11 04:50:47] openrec INFO: 160580
[2026/07/11 04:50:47] openrec INFO: 130975
[2026/07/11 04:50:47] openrec INFO: 117729
[2026/07/11 04:50:47] openrec INFO: 86849
[2026/07/11 04:50:47] openrec INFO: 75876
[2026/07/11 04:50:47] openrec INFO: 62836
[2026/07/11 04:50:47] openrec INFO: 251486
[2026/07/11 04:50:49] openrec INFO: train dataloader has 5689 iters
[2026/07/11 04:50:49] openrec INFO: Initialize indexs of datasets:/opt/data/private/data/benchmark_bctr/benchmark_bctr_test/document_test
[2026/07/11 04:50:49] openrec INFO: Initialize indexs of datasets:/opt/data/private/data/handwriting1/test
[2026/07/11 04:50:49] openrec INFO: Initialize indexs of datasets:/opt/data/private/data/benchmark_bctr/benchmark_bctr_test/scene_test
[2026/07/11 04:50:49] openrec INFO: Initialize indexs of datasets:/opt/data/private/data/benchmark_bctr/benchmark_bctr_test/web_test
[2026/07/11 04:50:53] openrec INFO: 0
[2026/07/11 04:50:53] openrec INFO: 27361
[2026/07/11 04:50:53] openrec INFO: 21063
[2026/07/11 04:50:53] openrec INFO: 17605
[2026/07/11 04:50:53] openrec INFO: 16078
[2026/07/11 04:50:53] openrec INFO: 12414
[2026/07/11 04:50:53] openrec INFO: 10839
[2026/07/11 04:50:53] openrec INFO: 9360
[2026/07/11 04:50:53] openrec INFO: 36469
The total number is approximately:1,096,238
However, in the original training log that I referred to, the corresponding bucket statistics are:1,092,292
The two logs differ by about 3,946 samples, and the bucket distribution is also noticeably different. As far as I understand, the batch size and the number of GPUs may affect the number of dataloader iterations per epoch, but they should not change the actual dataset size or the bucket sample statistics.Could I ask whether you know what might cause this difference? Could it be related to the LMDB version, data conversion process, or filtering rules during dataset construction?Thank you again for your help.
